src/cli.py

```diff
@@ -18,14 +18,26 @@ def log_messages(self, messages):
             self.writer(messages[message])
     
     def prompt_user(self, message=""):
-        return self.reader(message)
+        user_input = self.reader(message)
+        error_message = "It seems you may have entered some invalid input. Please try again:\n"
+        if self.validate_input(user_input):
+            return user_input
+        else:
+            return self.prompt_user(error_message)
```

Comment:
Whenever you use recursion, you run the risk of a stack overflow. Can you research ways to implement this that will make that scenario impossible?

Reply:
So leave out recursion completely? or alter this to prevent a stack overflow from accidently happening?

Reply:
I think you'll find that answer yourself in your research.